### PR TITLE
33 get userId from tocken

### PR DESCRIPTION
**Pull request by ihor-mk**

### CODE REVIEW

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        public async Task InvokeAsync(HttpContext context, IFirebaseAuthService firebaseAuthService)
+        {
+            var authToken = context.Request.Headers["Authorization"].ToString();
+
+            if (!string.IsNullOrEmpty(authToken) && authToken.StartsWith("Bearer "))
+            {
+                var idToken = authToken.Substring("Bearer ".Length);
+
+                try
+                {
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs:24]: We don't need try/catch we have Global Error Handler

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs (hunk 1 above)
Comment (ihor-mk) [reply backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs]: Remove this one

File: backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public FirebaseAuthMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context, IFirebaseAuthService firebaseAuthService)
+        {
+            var authToken = context.Request.Headers["Authorization"].ToString();
+
+            if (!string.IsNullOrEmpty(authToken) && authToken.StartsWith("Bearer "))
```
[the remark this answers is not in the document]
Comment (ihor-mk) [reply backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs:19]: Done

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/FirebaseAuthService.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            _context = context;
+        }
+        public long UserId { get; private set; }
+        
+        
+        public async Task SetUserId(string email)
+        {
+            var user =  await _context.Users
+                .Where(user => user.Email == email)
+                .FirstOrDefaultAsync();
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/FirebaseAuthService.cs:21]: You can merge these methods

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**